### DISCUSSION TITLE
How to use the AST viewer with jdk > 1.4 ?

### DISCUSSION THREAD

**Discussion by oowekyala** [pmd/pmd#3055] [Q&A] [answered]
When I use the `bgastviewer` packaged with pmd, I cannot use features introduced later than jdk 1.4 (annotations, lambdas, type inference for generics, etc.). I get the following error message on compilation : `Can't use annotations when running in JDK 1.4 mode!`

How to set it to a higher version ?

**Comment by jsotuyod** [comment, marked as answer]
From the Language menu you should pick a different Java version. It's currently limited to JDK 1.7, since it's long been unmaintained.

At some point we should merge both GUI tools for PMD (designer and bgastviewer), as stated [on the roadmap](https://github.com/pmd/pmd/wiki/Roadmap-and-future-directions#miscellaneous-changes). The designer is generally more complete, but the bgastviewer has a nicer codebase, and the AST is easier to navigate (although lots of properties are not currently visible).

I'd recommend you stick with the designer unless you have a very compelling reason.
